### PR TITLE
add more trips functionality and create edit trip

### PR DESCRIPTION
**Pull request by sandmich**

### CODE REVIEW

File: app/src/main/java/com/example/packpals/viewmodels/TripsPageViewModel.kt [hunk 1]
```diff
@@ -8,29 +8,39 @@ import androidx.lifecycle.viewModelScope
...
 
+    fun fetchPalsList(){
+        val userId = authRepo.getCurrentUID()
+        viewModelScope.launch {
+            val pal = userId?.let { palsRepo.fetchPal(it) }
+            if (pal != null) {
+                _palsList.value = listOf(Pal(pal.id,pal.name))
+            }
+            else{
+                _palsList.value = listOf(Pal("id1","not yuh"))
```
Comment (stephchi0) [inline app/src/main/java/com/example/packpals/viewmodels/TripsPageViewModel.kt:39]: yuh

File: app/src/main/java/com/example/packpals/models/Pal.kt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.example.packpals.models
 
 data class Pal(
-    val id: String? = null,
```
Comment (stephchi0) [inline app/src/main/java/com/example/packpals/models/Pal.kt:4]: prob better to leave as val but its ok good enough

**Review by stephchi0 — APPROVED** [review]

File: app/src/main/java/com/example/packpals/viewmodels/TripsPageViewModel.kt (hunk 1 above)
Comment (sandmich) [reply app/src/main/java/com/example/packpals/viewmodels/TripsPageViewModel.kt:39]: yuh

File: app/src/main/java/com/example/packpals/models/Pal.kt (hunk 2 above)
Comment (sandmich) [reply app/src/main/java/com/example/packpals/models/Pal.kt:4]: sgtm

**Pull request merged**